examples/quickrefl/core/editors/importdatawidgets/famousloader/importtextview.cpp

```diff
@@ -105,25 +84,52 @@ void ImportTextView::highlightCurrentLine()
     setExtraSelections(extraSelections);
 }
 
-//![cursorPositionChanged]
+//! Get the first visible block id
+int ImportTextView::getFirstVisibleBlockId()
+{
+    QTextCursor curs = QTextCursor(this->document());
+    curs.movePosition(QTextCursor::Start);
+    for(int i=0; i < this->document()->blockCount(); ++i)
+    {
+        QTextBlock block = curs.block();
 
-//![extraAreaPaintEvent_0]
+        QRect r1 = this->viewport()->geometry();
+        QRect r2 = this->document()->documentLayout()->blockBoundingRect(block).translated(
+                    this->viewport()->geometry().x(), this->viewport()->geometry().y() - (
+                        this->verticalScrollBar()->sliderPosition()
+                        ) ).toRect();
```

Comment:
Please [avoid long chains of function calls](https://en.wikipedia.org/wiki/Law_of_Demeter). Please simplify the statement by introducing well-named const variables.
```
const int current_slider_pos = this->verticalScrollBar()->sliderPosition();
```